manifests/versionlock.pp

```diff
@@ -1,39 +1,103 @@
 # @summary Locks package from updates.
 #
-# @example Sample usage
+# @example Sample usage on CentOS 7
 #   yum::versionlock { '0:bash-4.1.2-9.el6_2.*':
 #     ensure => present,
 #   }
 #
+# @example Sample usage on CentOS 8
+#   yum::versionlock { 'bash':
+#     ensure => present,
+#     version => '4.1.2',
+#     release => '9.el6',
+#     epoch   => '0',
+#     arch    => 'noarch',
+#   }
+#
+#
+# @example Sample usage on CentOS 7 with future method from 8
+#   yum::versionlock { 'bash':
+#     ensure  => present,
+#     future  => true,
+#     version => '4.1.2',
+#     release => '9.el6',
+#     epoch   => '0',
+#     arch    => 'noarch',
+#   }
+#
 # @param ensure
 #   Specifies if versionlock should be `present`, `absent` or `exclude`.
 #
+# @param version
+#   Version of the package if CentOS 8 mechanism is used.
+#
+# @param release
+#   Release of the package if CentOS 8 mechanism is used.
+#
+# @param arch
+#   Arch of the package if CentOS 8 mechanism is used.
+#
+# @param epoch
+#   Epoch of the package if CentOS 8 mechanism is used.
+#
 # @note The resource title must use the format
+#   By default on CentOS 7 the following format is used.
 #   "%{EPOCH}:%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}".  This can be retrieved via
 #   the command `rpm -q --qf '%{EPOCH}:%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}'.
 #   If "%{EPOCH}" returns as '(none)', it should be set to '0'.  Wildcards may
 #   be used within token slots, but must not cover seperators, e.g.,
 #   '0:b*sh-4.1.2-9.*' covers Bash version 4.1.2, revision 9 on all
 #   architectures.
+#   By default on CentOS 8 and newer the resource title  to just set the
+#   package name.
+#   If future is set to true on CentOS 7 then it behaves like CentOS 8
 #
 # @see http://man7.org/linux/man-pages/man1/yum-versionlock.1.html
 define yum::versionlock (
-  Enum['present', 'absent', 'exclude'] $ensure = 'present',
+  Enum['present', 'absent', 'exclude'] $ensure  = 'present',
+  Yum::RpmVersion                      $version     = '*',
+  Yum::RpmRelease                      $release     = '*',
+  Variant[Yum::RpmEpoch,Enum['*']]     $epoch       = 0,
+  Optional[Yum::RpmArch]               $arch        = undef,
```

Comment:
This also can be `*`

Reply:
Is `*` different to `undef` ?  

Are these the same in reality or different?

```
name-0:acd-def.*
```
is different to

```
name-0:abc-def
```

fine to add however.

Reply:
But Yum::RpmArch wouldn't allow *, right?

Reply:
I was questioning if needed since undef and `*` are equivalent but it makes sense to add in case they are not.